torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -41,6 +41,8 @@ using torch::distributed::autograd::DistAutogradContainer;
 namespace torch {
 namespace jit {
 
+std::recursive_mutex interpreter_lock;
```

Comment:
I don't think this lock is global enough. To be extra safe, shouldn't we only let one thread interpret TorchScript across all running computers in the world? I think you also need to setup a distributed service to be able to manage this distributed lock.